core/state/statedb_multicoin_test.go

```diff
@@ -13,38 +13,37 @@ import (
 	"github.com/ava-labs/libevm/core/rawdb"
 	"github.com/ava-labs/libevm/crypto"
 	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
 )
 
 func TestMultiCoinOperations(t *testing.T) {
 	s := newStateEnv()
 	addr := common.Address{1}
 	assetID := common.Hash{2}
 
-	root, _ := s.state.Commit(0, false)
-	s.state, _ = New(root, s.state.db, s.state.snaps)
+	root, err := s.state.Commit(0, false)
+	require.NoError(t, err, "committing state")
+	s.state, err = New(root, s.state.db, s.state.snaps)
+	require.NoError(t, err, "creating statedb")
 
 	s.state.AddBalance(addr, new(uint256.Int))
 
 	balance := s.state.GetBalanceMultiCoin(addr, assetID)
-	if balance.Cmp(big.NewInt(0)) != 0 {
-		t.Fatal("expected zero multicoin balance")
-	}
+	require.Equal(t, big.NewInt(0).String(), balance.String(), "expected zero multicoin balance")
 
 	s.state.AddBalanceMultiCoin(addr, assetID, big.NewInt(10))
 	s.state.SubBalanceMultiCoin(addr, assetID, big.NewInt(5))
 	s.state.AddBalanceMultiCoin(addr, assetID, big.NewInt(3))
 
 	balance = s.state.GetBalanceMultiCoin(addr, assetID)
-	if balance.Cmp(big.NewInt(8)) != 0 {
-		t.Fatal("expected multicoin balance to be 8")
-	}
+	require.Equal(t, big.NewInt(8).String(), balance.String(), "unexpected multicoin balance")
```

Comment:
```suggestion
	require.Equal(t, "8", balance.String(), "big.Int multicoin balance as string")
```

Reply:
That's not really the context, the problem is really an `unexpected multicoin balance string`. Otherwise it sounds like the string conversion is the problem 😉 

```suggestion
	require.Equal(t, big.NewInt(8).String(), balance.String(), "unexpected multicoin balance string")
```